bezier-rs/lib/src/lib.rs

```diff
@@ -122,57 +135,86 @@ impl Bezier {
 
 	pub fn get_handle1(&self) -> DVec2 {
 		match self.handles {
-			BezierHandles::Quadratic { handle } => {
-				handle
-			}
-			BezierHandles::Cubic { handle1, handle2: _ } => {
-				handle1
-			}
+			BezierHandles::Quadratic { handle } => handle,
+			BezierHandles::Cubic { handle1, handle2: _ } => handle1,
 		}
 	}
 
 	pub fn get_handle2(&self) -> Option<DVec2> {
 		match self.handles {
-			BezierHandles::Quadratic { handle: _ } => {
-				None
-			}
-			BezierHandles::Cubic { handle1: _, handle2 } => {
-				Some(handle2)
-			}
+			BezierHandles::Quadratic { handle: _ } => None,
+			BezierHandles::Cubic { handle1: _, handle2 } => Some(handle2),
 		}
 	}
 
-
 	pub fn get_points(&self) -> [Option<DVec2>; 4] {
 		match self.handles {
 			BezierHandles::Quadratic { handle } => [Some(self.start), Some(handle), Some(self.end), None],
 			BezierHandles::Cubic { handle1, handle2 } => [Some(self.start), Some(handle1), Some(handle2), Some(self.end)],
 		}
 	}
 
-	/// Return the length of the bezier curve
-	pub fn length() -> i32 {
-		0
-	}
-}
+	///  Calculate the x-value of a point on the curve based on the t-value provided
+	///  basis code based off of pseudocode found here: https://pomax.github.io/bezierinfo/#explanation
+	pub fn get_x_basis(&self, t: f64) -> f64 {
```

Comment:
Could we not make these two functions into one function that returns a `DVec2` to avoid duplication and then relevant code could do `point.x` and `point.y` when necessary?